client/post-editor/editor-title/index.jsx

```diff
@@ -40,6 +41,23 @@ export default React.createClass( {
 		};
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( isMobile() ) {
+			return;
+		}
+
+		// If next post is new, or the next site is different, focus title
+		if ( nextProps.isNew && ! this.props.isNew ||
+			( nextProps.isNew && ( this.props.site && nextProps.site ) && ( this.props.site.ID !== nextProps.site.ID ) )
+		) {
+			this.setState( {
+				isFocused: true
+			}, () => {
+				ReactDOM.findDOMNode( this.refs.titleInput ).focus();
```

Comment:
Do you think it'd be useful to add a `focus` method to `<FormTextInput />` so that it's easier to call as `this.refs.titleInput.focus();` ?